lib/active_merchant/billing/gateways/rapyd.rb

```diff
@@ -138,6 +139,15 @@ def add_address(post, creditcard, options)
         post[:address][:phone_number] = address[:phone].gsub(/\D/, '') if address[:phone]
       end
 
+      def valid_address?(address)
+        return false unless address
+
+        required_fields = %i[name address1 city state country zip]
+        missing_fields = required_fields.select { |field| address[field].nil? || address[field].to_s.strip.empty? }
```

Comment:
```suggestion
        missing_fields = required_fields.select { |field| address[field].to_s.strip.empty? }
```
As `nil` will be converted to empty string.